From fd72d6e970dce85f6f615dd3cbc3d038d0ab10c9 Mon Sep 17 00:00:00 2001
From: tbordaz <tbordaz@redhat.com>
Date: Mon, 20 Oct 2025 14:30:52 +0200
Subject: [PATCH] Issue 7032 - The new ipahealthcheck test
 ipahealthcheck.ds.backends.BackendsCheck raises CRITICAL issue (#7036)

Bug description:
	The bug fix #6966 adds a 'scanlimit' to one of the system
	index ('parentid'). So not all of them have such attribute.
	In healthcheck such attribute (i.e. key) can miss but
	the code assumes it is present

Fix description:
	Get 'parentid' from the dict with the proper routine
	(Thanks Florence Renaud for the debug/fix)

fixes: #7032

Reviewed by: Pierre Rogier and Simon Pichugin (thank you !)
---
 src/lib389/lib389/backend.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib389/lib389/backend.py b/src/lib389/lib389/backend.py
index 5c0bc6d40..3fef084ab 100644
--- a/src/lib389/lib389/backend.py
+++ b/src/lib389/lib389/backend.py
@@ -703,7 +703,7 @@ class Backend(DSLdapObject):
                             reindex_attrs.add(attr_name)
 
                     # Check fine grain definitions for parentid ONLY
-                    expected_scanlimit = expected_config['scanlimit']
+                    expected_scanlimit = expected_config.get('scanlimit')
                     if (attr_name.lower() == "parentid") and expected_scanlimit and (len(actual_scanlimit) == 0):
                             discrepancies.append(f"Index {attr_name} missing fine grain definition of IDs limit: {expected_mr}")
                             # Add the missing scanlimit
-- 
2.52.0

